src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -423,7 +422,19 @@ namespace winrt::Microsoft::Terminal::Control::implementation
                     }
                 }
 
-                _searchBox->SetFocusOnTextbox();
+                if (!_searchBox->IsLoaded())
+                {
+                    _revokers.SearchBoxLoaded = _searchBox->Loaded(winrt::auto_revoke, [this](auto&&, auto&&) {

```

Comment:
TermControl has to wait until SearchBox's SizeChanged event fires. This is where we set the initial values for our content clip rect and entry animation point. SizeChanged fires just before Loaded. Technically, it also has to wait for SearchBox's Loaded event because that puts it into the default "Hidden" state.

```cpp
// in SearchBoxControl.cpp

_initialLoadedRevoker = Loaded(winrt::auto_revoke, [this](auto&&, auto&&) {
    VisualStateManager::GoToState(*this, L"Hidden", false);
    _initialLoadedRevoker.revoke();
});

SizeChanged([this](auto&&, auto&&) {
    _UpdateSizeDependents();
});
```

The chain of events looks something like this:

```txt
SearchBox ctor -> SearchBox SizeChanged handler -> SearchBox Loaded handler -> TermControl Loaded handler
```

Reply:
For extra lolz a slide-in transition in plain CSS: [A minor CSS flex.](https://flems.io/#0=N4IgzgpgNhDGAuEAmIBcIB0ALeBbKIANCAGYCWMYaA2qAHYCGuEamO+RIsA9nYn6wA8AQgAiAeQDCAFQCaABQCiAAnZQAfAB06gtcqgM6AcwC8mkBDrmtOgEbckATxvLlgpGQBuL127AAHQ3UoXAZuZQAPJihBAHoAoO1XOI9vbTj7Jxs4tRtOSBgEMl4qdAAWVABGAE4QAF9CeiYWdAwAKypiHj5LeFZ6xpBGZlYMWDBOrl5+PvRMx2VgJOVQgCcjMjpUZUqIXABuZdsGWABrI1XuAFc6JABaHihuVe2kBlXTsANEC4ZHQ7orhI022ZT2yjAjjAiFwdyuZABrkez22AHcsGREAC6tptKlFstuJ4IKsSE9UdsMUgkJZsbi6AlAUtAcoPAEDI5tpsoJsIHdbE8zoiVu8NltlAAGDAAVj2wsC1M2Rm2UsqACZZbhJTK5UcTudLjckK93qdQkZegxhYYyKF4MVxQBmMDKTbkOiYiDKBhQRCrRiICE8mmbaTcfyyOlWOgAAVOEEcJFWzRdXzIIboYYjBJZaolAFIc75lPBkwzgatcNtS4YvgxELIABSwH2wRt3SoSgvKO7azUASn7yku8HrEEbEppRn7wpxLIAHN3mcWa+XnlWS2W6w2J0OR2PGwB2PNTmfLOdz-LQOD2kqsNWH1AS+oAXTqQA)